### PR TITLE
Pal 642 data service prepopulation

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserConstructorNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package uk.gov.gchq.palisade.service.data.exception;
+
+/**
+ * A {@link RuntimeException} that is thrown when incorrect values are used for the service Prepopulation.
+ */
+public class SerialiserConstructorNotFoundException extends RuntimeException {
+
+    /**
```
Comment (m78233) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserConstructorNotFoundException.java:24]: Would have expected the other two constructor methods that we look like we use by convention:
SerialiserConstructorNotFoundException(final String message)
SerialiserConstructorNotFoundException(final Throwable cause)

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserInitialisationException.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+public class SerialiserInitialisationException extends RuntimeException {
+
+    /**
+     * Constructs a new {@link SerialiserInitialisationException} with the specified detail message and cause.
+     *
+     * @param message   a {@link String} value detailing the error
+     * @param cause     a {@link Throwable} that caused the iss
+     */
+    public SerialiserInitialisationException(final String message, final Throwable cause) {
```
Comment (m78233) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserInitialisationException.java:48]: Similar question here as with SerialiserConstructorNotFoundException,  consider the other constructors.

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ */
+public class SerialiserNotFoundException extends RuntimeException {
+
+    /**
+     * Constructs a new {@link SerialiserNotFoundException} with the specified detail message and cause.
+     *
+     * @param message   a {@link String} value detailing the error
+     * @param cause     a {@link Throwable} that caused the iss
+     */
+    public SerialiserNotFoundException(final String message, final Throwable cause) {
```
Comment (m78233) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserNotFoundException.java:48]: Similar question here as with SerialiserConstructorNotFoundException, consider the other constructors.

**Review by developer01189998819991197253 — DISMISSED** [review]

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataController.java [hunk 4]
```diff
@@ -53,10 +67,24 @@ public DataController(final DataService service) {
     @PostMapping(value = "/addSerialiser", consumes = "application/json", produces = "application/json")
```
Comment (ac74475) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataController.java]: given we are now using pre population, I don't think we need to have a public rest end point for adding the serialisers, which means we can remove the addSerialiserRequest class. I would leave the data service interface addSerialiser method though

**Review by ac74475 — CHANGES_REQUESTED** [review]
One small change to remove some unnecessary parts

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserConstructorNotFoundException.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     * Constructs a new {@link SerialiserConstructorNotFoundException} with the specified detail message and cause.
```
Comment (developer01189998819991197253) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserConstructorNotFoundException.java]: change it so javadoc is specific for each constructor. 
One for message
One for cause
one for both, same in SerlaiserInitialisationException and SerialiserNotFoundException

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserInitialisationException.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+package uk.gov.gchq.palisade.service.data.exception;
+
+/**
+ * A {@link RuntimeException} that is thrown when incorrect values are used for the service Prepopulation.
+ */
+public class SerialiserInitialisationException extends RuntimeException {
+
+    /**
+     * Constructs a new {@link SerialiserInitialisationException} with the specified detail message and cause.
```
Comment (developer01189998819991197253) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserInitialisationException.java]: change it so javadoc is specific for each constructor. 
One for message
One for cause
one for both, same in SerlaiserInitialisationException and SerialiserNotFoundException

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserNotFoundException.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+     * Constructs a new {@link SerialiserNotFoundException} with the specified detail message and cause.
+     *
+     * @param cause     a {@link Throwable} that caused the error
+     */
+    public SerialiserNotFoundException(final Throwable cause) {
+        super(cause);
+    }
+
+    /**
+     * Constructs a new {@link SerialiserNotFoundException} with the specified detail message and cause.
```
Comment (developer01189998819991197253) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserNotFoundException.java]: change it so javadoc is specific for each constructor. 
One for message
One for cause
one for both, same in SerlaiserInitialisationException and SerialiserNotFoundException

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserNotFoundException.java (hunk 7 above)
Comment (pd104923) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserNotFoundException.java]: Fixed

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserInitialisationException.java (hunk 6 above)
Comment (pd104923) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserInitialisationException.java]: Fixed

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserConstructorNotFoundException.java (hunk 5 above)
Comment (pd104923) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/exception/SerialiserConstructorNotFoundException.java]: Fixed

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataController.java (hunk 4 above)
Comment (pd104923) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataController.java]: The AddSerialiserRequest has been deleted and the rest end point has been removed.

Comment (pd104923) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/web/DataController.java]: These have been removed, required some changes to the integration tests.

**Review by m78233 — DISMISSED** [review]

**Review by ac74475 — DISMISSED** [review]

**Review by developer01189998819991197253 — DISMISSED** [review]

**Review by dev930018 — APPROVED** [review]

**Review by dev930018 — CHANGES_REQUESTED** [review]
You will need changes in all resource services to use fully qualified classpath for the type
For the simple resource service, i would recommend using java.lang.String as the type
To avoid having to change anything for hadoop, you should just be able to rename the employee_fileXX.avro to uk.gov.....Employee_fileXX.avro
This should be done in all appropriate places

**Comment by ac74475** [comment]
I don't like the idea of changing the name of the files, I would rather have a lookup map of known types within the HadoopResourceDetails class which can map employee to the fully qualified domain name for the Employee class. I agree with changing the currently hardcoded value of 'txt' too 'java.lang.String' in the SimpleResourceService.

**Review by dev930018 — APPROVED** [review]

**Review by m78233 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**